vault/resource_kubernetes_auth_backend_config.go

```diff
@@ -0,0 +1,237 @@
+				Description: "Host must be a host string, a host:port pair, or a URL to the base of the Kubernetes API server.",
+			},
+			"kubernetes_ca_cert": {
+				Type:        schema.TypeString,
+				Description: "PEM encoded CA cert for use by the TLS client used to talk with the Kubernetes API.",
+				Optional:    true,
+				ForceNew:    true,
+				Default:     "",
+			},
+			"token_reviewer_jwt": {
+				Type:        schema.TypeString,
+				Description: "A service account JWT used to access the TokenReview API to validate other JWTs during login. If not set the JWT used for login will be used to access the API.",
+				ForceNew:    true,
+				Default:     "",
+				Optional:    true,
+			},
+			"pem_keys": {
+				Type:        schema.TypeList,
+				Elem:        &schema.Schema{Type: schema.TypeString},
+				Description: "Optional list of PEM-formatted public keys or certificates used to verify the signatures of Kubernetes service account JWTs. If a certificate is given, its public key will be extracted. Not every installation of Kubernetes exposes these keys.",
+				ForceNew:    true,
+				Optional:    true,
+			},
+			"backend": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "Unique name of the kubernetes backend to configure.",
+				ForceNew:    true,
+				Default:     "kubernetes",
+				// standardise on no beginning or trailing slashes
+				StateFunc: func(v interface{}) string {
+					return strings.Trim(v.(string), "/")
+				},
+			},
+		},
+	}
+}
+
+func kubernetesAuthBackendConfigPath(backend string) string {
+	return "auth/" + strings.Trim(backend, "/") + "/config"
+}
+
+func kubernetesAuthBackendConfigCreate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*api.Client)
+
+	backend := d.Get("backend").(string)
+
+	path := kubernetesAuthBackendConfigPath(backend)
+	log.Printf("[DEBUG] Writing Kubernetes auth backend config %q", path)
+
+	data := map[string]interface{}{}
+
+	if v, ok := d.GetOk("kubernetes_ca_cert"); ok {
+		kubernetesCAcert := v.(string)
+		if len(kubernetesCAcert) > 0 {
+			data["kubernetes_ca_cert"] = kubernetesCAcert
+		}
+	}
+
+	if v, ok := d.GetOk("token_reviewer_jwt"); ok {
+		tokenReviewerJWT := v.(string)
+		if len(tokenReviewerJWT) > 0 {
+			data["token_reviewer_jwt"] = tokenReviewerJWT
+		}
+	}
+
+	if v, ok := d.GetOkExists("pem_keys"); ok {
+		var pemKeys []string
+		for _, pemKey := range v.([]interface{}) {
+			pemKeys = append(pemKeys, pemKey.(string))
+		}
+		data["pem_keys"] = strings.Join(pemKeys, ",")
+	}
+	data["kubernetes_host"] = d.Get("kubernetes_host").(string)
+
+	_, err := client.Logical().Write(path, data)
+	if err != nil {
+		return fmt.Errorf("error writing Kubernetes auth backend config %q: %s", path, err)
+	}
+
+	d.SetId(path)
+	// NOTE: Since reading the auth/<backend>/config does
+	// not return the `token_reviewer_jwt`,
+	// set it from data after successfully storing it in Vault.
+	d.Set("token_reviewer_jwt", data["token_reviewer_jwt"])
```

Comment:
Note to reviewers that this is probably dangerous.

It seems like the only way to get the `token_reviewer_jwt` in the terraform state is to set it after successfully writing the data to Vault.